auctionapp/src/test/java/com/internship/auctionapp/service/BidServiceTest.java

```diff
@@ -0,0 +1,180 @@
+package com.internship.auctionapp.service;
+
+import com.internship.auctionapp.dto.BidDto;
+import com.internship.auctionapp.entity.Bid;
+import com.internship.auctionapp.entity.Item;
+import com.internship.auctionapp.entity.User;
+import com.internship.auctionapp.exception.BadRequestException;
+import com.internship.auctionapp.repository.BidRepository;
+import com.internship.auctionapp.repository.ItemRepository;
+import com.internship.auctionapp.service.impl.BidServiceImpl;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentMatchers;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.modelmapper.ModelMapper;
+
+import java.time.LocalDateTime;
+import java.time.Month;
+import java.util.Optional;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+@ExtendWith(MockitoExtension.class)
+class BidServiceTest {
+
+    @Mock
+    ItemRepository itemRepository;
+    @Mock
+    ModelMapper mapper = new ModelMapper();
+    @Mock
+    private BidRepository bidRepository;
+    @InjectMocks
+    private BidServiceImpl bidService;
+
+    @BeforeEach
+    void setUp() {
+    }
+
+    @AfterEach
+    void tearDown() {
+    }
+
+
+    @Test
+    void saveNewBid_amountLowerThanStartPrice_badRequestException() {
+        BidDto bidDto = BidDto.builder()
+                .id(UUID.randomUUID())
+                .itemId(UUID.randomUUID())
+                .userId(UUID.randomUUID())
+                .amount(5)
+                .build();
```

Comment:
Move this below, private method getBid(), getItem()...., things that you use for multiple tests. You can then set specific field in test